pkg/controllers/clusterresourceplacement/controller_integration_test.go

```diff
@@ -534,6 +662,596 @@ var _ = Describe("Test ClusterResourcePlacement Controller", func() {
+							Conditions: []metav1.Condition{
+								{
+									Status: metav1.ConditionTrue,
+									Type:   string(placementv1beta1.ResourceScheduledConditionType),
+									Reason: condition.ScheduleSucceededReason,
+								},
+								{
+									Status: metav1.ConditionTrue,
+									Type:   string(placementv1beta1.ResourceRolloutStartedConditionType),
+									Reason: condition.RolloutStartedReason,
+								},
+								{
+									Status: metav1.ConditionTrue,
+									Type:   string(placementv1beta1.ResourceOverriddenConditionType),
+									Reason: condition.OverriddenSucceededReason,
+								},
+								{
+									Status: metav1.ConditionTrue,
+									Type:   string(placementv1beta1.ResourceWorkSynchronizedConditionType),
+									Reason: condition.WorkSynchronizedReason,
+								},
+								{
+									Status: metav1.ConditionTrue,
+									Type:   string(placementv1beta1.ResourcesDiffReportedConditionType),
+									Reason: condition.DiffReportedStatusTrueReason,
+								},
+							},
+						},
+					},
+				},
+			}
+			gotCRP = retrieveAndValidateClusterResourcePlacement(testCRPName, wantCRP)
+
+			By("Ensure placement complete metric was emitted with isCompleted True")
+			checkPlacementCompleteMetric(customRegistry, testCRPName, true, 2)
+
+			By("Ensure placement status metric was emitted")
+			status := map[string]string{
+				string(placementv1beta1.ClusterResourcePlacementScheduledConditionType):      string(corev1.ConditionUnknown),
+				string(placementv1beta1.ClusterResourcePlacementRolloutStartedConditionType): string(corev1.ConditionUnknown),
+				string(placementv1beta1.ClusterResourcePlacementDiffReportedConditionType):   string(corev1.ConditionUnknown),
+			}
+			checkPlacementStatusMetric(customRegistry, crp, status)
+		})
+	})
+})
+
+func checkPlacementCompleteMetric(registry *prometheus.Registry, crpName string, complete bool, numOfMetrics int) {
+	metricFamilies, err := registry.Gather()
+	Expect(err).Should(Succeed())
+	var placementCompleteMetrics []*prometheusclientmodel.Metric
+	for _, mf := range metricFamilies {
+		if mf.GetName() == "fleet_workload_placement_complete_last_timestamp_seconds" {
+			placementCompleteMetrics = mf.GetMetric()
+		}
+	}
+	// we only expect one metric with 2 label
+	Expect(len(placementCompleteMetrics)).Should(Equal(numOfMetrics))
+
+	for _, emittedMetric := range placementCompleteMetrics {
+		metricLabels := emittedMetric.GetLabel()
+		Expect(len(metricLabels)).Should(Equal(2))
+		for _, label := range metricLabels {
+			if label.GetName() == "isComplete" {
+				Expect(label.GetValue()).Should(Equal(complete))
+			}
+			if label.GetName() == "name" {
+				Expect(label.GetValue()).Should(Equal(crpName))
+			}
+		}
+		Expect(emittedMetric.GetGauge().GetValue()).ShouldNot(Equal(float64(0)))
+	}
+}
+
+func checkPlacementStatusMetric(registry *prometheus.Registry, crp *placementv1beta1.ClusterResourcePlacement, statuses map[string]string) {
```

Comment:
please refer to my sample code, https://github.com/Azure/fleet/pull/1110/commits/7abce3f11d9c2376428bde9414cc5eed450da04a